### ISSUE TITLE
Fetching FileHandleStatus first and FileStream after leads to NPE

### ISSUE THREAD

**Issue by rnorden** [lokra/seaweedfs-client#2]
Whenever you request a file and try to get the FileHandleStatus and after that the FileStream to i.e. compose a response with both combined, there will be a NPE thrown at org.lokra.seaweedfs.core.Connection.fetchStreamCacheByRequest(Connection.java:328).

```java
public class SeaweedFile {
	@JsonProperty
	public FileHandleStatus status;
	@JsonProperty
	public String data;
}

@GET
@Path("{id}")
public Response read(@PathParam("id") String fileID) {
	Response response = null;
	FileTemplate template = new FileTemplate(fileSource.getConnection());
	try {
		SeaweedFile file = new SeaweedFile();
		file.setStatus(template.getFileStatus(fileID)); // first get the FileStatus  *X
		
		StreamResponse fileStream = template.getFileStream(fileID); // *Y
		InputStream is = fileStream.getInputStream();
		file.setData(new String(ByteStreams.toByteArray(is)));
               // do return stuff ...
       } ...
}
```

If you switch lines marked with *X and *Y, the request will return smoothly.

**Comment by ChihoSin** [comment]
@rnorden 

I think the problem is caused by HttpClient cache policy, this client will using HEAD request when you get the file status (no content, but save at response cache), and then client fetching content using GET request, but HttpClient found this request at cache.

I have been busy recently and I hope someone can help me fix this problem.

**Comment by ChihoSin** [comment]
fixed #3 at version 0.7.3.RELEASE.
